lsp-types/src/Language/LSP/Protocol/Types/CustomInstances.hs

```diff
@@ -0,0 +1,27 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Language.LSP.Protocol.Types.CustomInstances where
+
+import           Language.LSP.Protocol.Internal.Types
+import           Data.Default
+import           Data.Hashable
+import           Data.Semigroup                ()
+
+instance Semigroup WorkspaceEdit where
+  (WorkspaceEdit a b c) <> (WorkspaceEdit a' b' c') = WorkspaceEdit (a <> a') (b <> b') (c <> c')
+instance Monoid WorkspaceEdit where
+  mempty = WorkspaceEdit Nothing Nothing Nothing
+
+instance Hashable Location
+instance Hashable Range
+instance Hashable Position
```

Comment:
Hmm, this one we should maybe just derive for everything?

Reply:
I guess it depends on how badly it will affect compile time. It is only used with these types in the codebase.

Reply:
I think the compile time is still not too bad, thanks to it being very parallel.